fetching/tarball-fetcher/src/remoteTarballFetcher.ts

```diff
@@ -157,27 +157,43 @@ export function createDownloader (
           ? opts.onProgress
           : undefined
         let downloaded = 0
+        const chunks: Buffer[] = []
         res.body!.on('data', (chunk: Buffer) => {
+          chunks.push(chunk)
           downloaded += chunk.length
           if (onProgress != null) onProgress(downloaded)
```

Comment:
You may also want to look into performance tuning of your default progress handler. When I profiled several seconds of main thread were spent inside the progress handler.

Reply:
I have added throttling to progress reporting, so it should do a lot less processing.